controllers/drplacementcontrol_controller.go

```diff
@@ -1750,65 +1901,235 @@ func (s ObjectStorePVDownloader) DownloadPVs(ctx context.Context, r client.Reade
 }
 
 func (d *DRPCInstance) advanceToNextDRState() {
-	var nextState rmn.DRState
+	lastDRState := d.getLastDRState()
+	nextState := lastDRState
 
-	switch d.instance.Status.LastKnownDRState {
-	case rmn.DRState(""):
-		nextState = rmn.Initial
+	switch lastDRState {
+	case rmn.Deploying:
+		nextState = rmn.Deployed
 	case rmn.FailingOver:
 		nextState = rmn.FailedOver
 	case rmn.FailingBack:
 		nextState = rmn.FailedBack
 	case rmn.Relocating:
 		nextState = rmn.Relocated
-	case rmn.Initial:
+	case rmn.Deployed:
 	case rmn.FailedOver:
 	case rmn.FailedBack:
 	case rmn.Relocated:
-	default:
-		nextState = rmn.DRState("")
 	}
 
 	d.setDRState(nextState)
 }
 
-func (d *DRPCInstance) resetDRState() {
-	d.log.Info("Resetting last known DR state", "lndrs", d.instance.Status.LastKnownDRState)
+func (d *DRPCInstance) setDRState(nextState rmn.DRState) {
+	if d.instance.Status.Phase != nextState {
+		d.log.Info(fmt.Sprintf("Phase: Current '%s'. Next '%s'",
+			d.instance.Status.Phase, nextState))
 
-	d.setDRState(rmn.DRState(""))
+		d.instance.Status.Phase = nextState
+		d.updateConditions()
+		d.needStatusUpdate = true
+	}
 }
 
-func (d *DRPCInstance) setDRState(nextState rmn.DRState) {
-	if d.instance.Status.LastKnownDRState != nextState {
-		d.log.Info(fmt.Sprintf("LastKnownDRState: Current State '%s'. Next State '%s'",
-			d.instance.Status.LastKnownDRState, nextState))
+func (d *DRPCInstance) updateConditions() {
+	d.log.Info(fmt.Sprintf("Current Conditions '%v'", d.instance.Status.Conditions))
 
-		d.instance.Status.LastKnownDRState = nextState
-		d.needStatusUpdate = true
+	for _, condType := range []string{rmn.ConditionAvailable, rmn.ConditionReconciling} {
+		condition := d.newCondition(condType)
+
+		idx, _ := GetDRPCCondition(&d.instance.Status, condType)
+		if idx == -1 {
+			d.instance.Status.Conditions = append(d.instance.Status.Conditions, *condition)
+		} else {
+			d.instance.Status.Conditions[idx] = *condition
+		}
+	}
+
+	d.log.Info(fmt.Sprintf("Updated Conditions '%v'", d.instance.Status.Conditions))
+}
+
+func (d *DRPCInstance) newCondition(condType string) *metav1.Condition {
+	return &metav1.Condition{
+		Type:   condType,
+		Status: d.getConditionStatus(condType),
+		// ObservedGeneration: d.getObservedGeneration(condType),
+		LastTransitionTime: metav1.Now(),
+		Reason:             d.getConditionReason(condType),
+		Message:            d.getConditionMessage(condType),
+		ObservedGeneration: d.instance.Generation,
```

Comment:
@BenamarMk apart from the DRPC `Status` omitmempty fix, I was playing with status generations here, nothing concrete, but you can take a look.